server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java

```diff
@@ -124,20 +127,28 @@ public void run() {
           interruptFlag.set(false);
 
         } catch (IOException e) {
-          log.warn("lookup failed for tablet " + entry.getKey(), e);
+          log.warn("lookup failed for tablet " + extent, e);
           throw new RuntimeException(e);
         }
 
         bytesAdded += lookupResult.bytesAdded;
 
         if (lookupResult.unfinishedRanges.isEmpty()) {
-          fullScans.add(entry.getKey());
+          fullScans.add(extent);
+          // if this extent was previously saved, but now completed then reset these values
+          if (partScan != null && partScan.equals(extent)) {
+            partScan = null;
+            partNextKey = null;
+            partNextKeyInclusive = false;
+          }
         } else {
           if (lookupResult.closed) {
-            failures.put(entry.getKey(), lookupResult.unfinishedRanges);
+            failures.put(extent, lookupResult.unfinishedRanges);
           } else {
-            session.queries.put(entry.getKey(), lookupResult.unfinishedRanges);
```

Comment:
How were we not always getting concurrent modification exceptions here?  Weird

Reply:
CME occurred the next time iter.next() was called, so the while condition would have to allow one more execution.  My theories are:  1) low number of yields 2) only one extent 3) yield happens on last extent 4) timeout in while coincides with yield